docs/source/tracking.rst

```diff
@@ -652,13 +652,7 @@ There are also two ways to authenticate to HDFS:
 Most of the cluster contest settings are read from ``hdfs-site.xml`` accessed by the HDFS native
 driver using the ``CLASSPATH`` environment variable.
 
-Optionally you can select a different version of the HDFS driver library using:
```

Comment:
Can we still leave this as an option? Just mention that if you use anything else than libhdfs it will not work with newer pyarrow?

Reply:
Thanks for the review. 

If we leave the option for hdfs3 it would introduce code like:
```
if pyarrow > 0.16:
    connected = pa.hdfs.connect(host=host or 'default',
                                port=port or 0,
                                user=kerberos_user,
                                kerb_ticket=kerb_ticket,
                                extra_conf=extra_conf)
else:
    connected = pa.hdfs.connect(host=host or 'default',
                                port=port or 0,
                                user=kerberos_user,
                                driver=driver,
                                kerb_ticket=kerb_ticket,
                                extra_conf=extra_conf)
```

It would make the code less clean.

Is anyone using hdfs3 for real ? What about removing and if somebody complains discuss again to eventually put it back ?

At Criteo we were using hdfs3 with https://github.com/dask/hdfs3 for a long time. It is pretty nice but it is unmaintained and doesn't support viewfs for example. 

I also made some tests and the throughput of libhdfs vs hdfs3 is mostly the same.
The only reason for hdfs is that it doesn't need the JVM mounted via JNI. So 'time to first byte' when starting a new python interpreter is better. But that is really is the only reason.

So we decided to move to libdhfs via arrow's implementation.

In a next PR I would also like to commit https://github.com/apache/arrow/pull/5957 to add viewfs support to mlflow (which only works for libhdfs and not for hdfs3)

Also nobody noticed so far that hdfs3 is broken in later pyarrow versions.

Reply:
Those are good points. Can you include a summary of this conversation in the release notes section of the PR description? I.e. say something like - in order to maintain newer pyarrow version we have dropped support for hdfs3 driver.